src/sentry/api/fields/actor.py

```diff
@@ -23,9 +24,14 @@ def from_actor_id(cls, actor_id):
         if actor_id.isdigit():
             return Actor(int(actor_id), User)
 
-        type_name, _, id = actor_id.partition(':')
+        if u':' in actor_id:
```

Comment:
I don't think that this check is safe. I assume `:` is a valid character within an email address or legacy username?

I'd argue that we should change this instead to explicitly look for `team:` or `user:` prefix instead of this logic that I spit out if we need to handle this type of input now.

Maybe at this point, we also allow `#foo` for team name? Instead of `team:1` ? Just to make this generically handle all types of input.

Reply:
I could handle team name like that but i wish this endpoint accepted less types of input, not more. if you think it's a good idea i can do it

Reply:
Yeah, I don't really think it's great.